### DISCUSSION TITLE
How to configure keycloak auth?

### DISCUSSION THREAD

**Discussion by perrfect** [alephdata/aleph#3686] [Q&A] [answered]
Hello.
I have Aleph and Keycloak in docker compose on the same machine and want to configure keycloak auth, but it is not working.
**Aleph** run on the **port 8000**, **Keycloak** run on the **port 80** on the same machine.

Could you please help me with that?

My keycloak config:

**Client:** aleph
**Client ID:** aleph
**Enable:** On
**Client Protocol:** openid-connect
**Root URL:** http://192.168.1.188:8000
**Home URL:** http://192.168.1.188:8000
**Valid redirect URIs:** http://192.168.1.188:8000/*
**Web origins:** http://192.168.1.188:8000
**Admin URL:** http://192.168.1.188:8000
**Client authentication:** On
**Authorization:** On
**Authentication flow:** Standard flow, Direct access grants

Aleph env conf:
```
ALEPH_OAUTH = true
ALEPH_OAUTH_KEY = aleph
ALEPH_OAUTH_SECRET = some_secret_here
ALEPH_OAUTH_METADATA_URL = http://192.168.1.188/auth/realms/aleph/openid-connect/auth
```
 But I'm getting the error:

### In browser:

Redirect to URL: `http://192.168.1.188:8000/api/2/session/oauth?next=%2F`
**Internal Server Error**

### In docker compose logs:

```
[DEBUG] GET /api/2/session/oauth
[ERROR] Error handling request /api/2/sessions/oauth?next=%2F
...
RuntimeError: Missing "authorize_url" value
```
Maybe my Aleph or Keycloak configurations are wrong? I didn't find any information about how to connect Keycloak to Aleph.
Please help.

**Comment by tillprochaska** [comment, marked as answer]
Hi! Thanks for opening this discussion thread and sorry for the late response. Anyone trying to setup Keycloak as an identity provider in the future might find this new documentation article helpful: https://docs.aleph.occrp.org/developers/how-to/operations/identity-provider/#keycloak
